src/kudu/consensus/raft_consensus.cc

```diff
@@ -3761,6 +3761,109 @@ Status RaftConsensus::GetVoterDistribution(std::map<std::string, int32> *vd) con
   return cmeta_->voter_distribution(vd);
 }
 
+Status RaftConsensus::ChangeQuorumType(QuorumType type) {
```

Comment:
I put my concerns about voter_distribution in the other commit.
Neither ChangeQuorumType or SetPeerQuorumIds is altering the existing
voter distribution. This is causing a skew between some maps being based
on quorum id and some maps being based on voter distribution and we are
reading it directly in Watermarking code. How are we going to handle this

Reply:
GetVoterDistributionForQuorumId() in quorum_util.cc is the function to handle this. In QuorumId type, it set vd default to 3, and overrides by VD provided by users. In Region type, it directly uses VD provided by users. In dynamic mode, it is supposed to call this function.